src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1128,7 +1128,7 @@ private void CheckRestrictedTypeReceiver(BoundExpression expression, CSharpCompi
 
             if (!gotError)
             {
-                gotError = !CheckInvocationArgMixing(
+                CheckInvocationArgMixing(
```

Comment:
It wasn't clear to me why we no longer update 'gotError' here.

Reply:
If the `BoundCall` returned from binding the `Deconstruct()` method has errors, the caller, `MakeDeconstructionConversion()` will avoid setting inferred types for deconstructed variables and will result in type inference errors. But `CheckInvocationArgMixing()` should only affect which ref safety diagnostics are reported for the method call, it should not affect whether the `BoundCall` has errors, or the resulting type inference.